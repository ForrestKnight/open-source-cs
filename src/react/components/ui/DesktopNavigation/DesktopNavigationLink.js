/* Packages */
import React from 'react';
import {NavLink} from 'react-router-dom';


/* Component */
export class DesktopNavigationLink extends React.Component {
    constructor(props) {
        super(props);
    };

    render() {
        return (
            <li className="DesktopNavigationLink">
                {this.props.internal == true ?
                    /* Internal link */
                    <NavLink className="DesktopNavigationLink__link"
                        to={this.props.href}
                    >
                        {this.props.text}
                    </NavLink>
                    :

                    /* External link */
                    <a className="DesktopNavigationLink__link"
                        href={this.props.href}
                    >
                        {this.props.text}
                    </a>
                }

                {/* Sublink menu */}
                {this.props.sublinks &&
                    <ul className="DesktopNavigationLink__sublink_list">
                        {this.props.sublinks.map((sublink) => (
                            <li className="DesktopNavigationLink__sublink"
                                key={sublink.text}
                            >
                                {sublink.internal == true ?
                                    /* Internal sublink */
                                    <NavLink className="DesktopNavigationLink__sublink_link"
                                        to={sublink.href}
                                    >
                                        {sublink.text}
                                    </NavLink>
                                    :

                                    /* External sublink */
                                    <a className="DesktopNavigationLink__sublink_link"
                                        href={sublink.href}
                                    >
                                        {sublink.text}
                                    </a>
                                }
                            </li>
                        ))}
                    </ul>

                }
            </li>
        );
    };
};


export default DesktopNavigationLink;