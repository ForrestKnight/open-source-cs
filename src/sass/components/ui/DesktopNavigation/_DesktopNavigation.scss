#DesktopNavigation {
    display: none;

    @media #{$desktop_breakpoint} {
        display: block;
    }
}


.DesktopNavigation__nav {
    position: fixed;
    right: 1.5em;
    top: 2em;
    z-index: 8;
}


.DesktopNavigation__list {
    padding: 0;
    list-style-type: none;
    display: flex;
    flex-direction: row;
    margin-top: 0;
    margin-bottom: 0;
    width: calc(100vw - 120px);
    flex-wrap: wrap;
    justify-content: flex-end;
}
