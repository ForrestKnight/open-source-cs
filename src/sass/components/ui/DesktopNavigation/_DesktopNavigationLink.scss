.DesktopNavigationLink {
    transition: 100ms;
}


/* 'Active' state */
.DesktopNavigationLink:hover {
    .DesktopNavigationLink__link {
        border-bottom: 2px solid $primaryColor;
        color: $black;
    }

    .DesktopNavigationLink__sublink_list {
        height: auto;
    }
}


.DesktopNavigationLink__link {
    color: $primaryColor;
    text-decoration: none;
    transition: 100ms;
    display: block;
    text-align: center;
    margin: 0 10px;
    padding: 0 10px;
}


.DesktopNavigationLink__sublink_list {
    list-style-type: none;
    overflow: hidden;
    background: $shadow;
    padding: 0;
    height: 0;
    z-index: 8;
}


.DesktopNavigationLink__sublink {
    &:not(:last-child) {
        border-bottom: 1px solid $primaryColor_light;
    }
}


.DesktopNavigationLink__sublink_link {
    color: $white;
    text-decoration: none;
    display: block;
    padding: 8px;
    text-align: center;
    font-size: $sFont;
    
    &:hover {
        background: $shadow_thick;
    }
}