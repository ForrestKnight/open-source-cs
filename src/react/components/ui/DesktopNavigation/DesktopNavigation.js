/* Packages */
import React from 'react';


/* Components */
import DesktopNavigationLink from './DesktopNavigationLink';


/* Component */
export class DesktopNavigation extends React.Component {
    constructor(props) {
        super(props);
    };

    render() {
        return (
            <section id="DesktopNavigation">
                <nav className="DesktopNavigation__nav">
                    <ul className="DesktopNavigation__list">
                        {/* Links */}
                        {this.props.links &&
                            this.props.links.map((link) => (
                                <DesktopNavigationLink 
                                    key={link.text}
                                    {...link}
                                />
                            ))
                        }                 
                    </ul>
                </nav>
            </section>
        );
    };
};


export default DesktopNavigation;