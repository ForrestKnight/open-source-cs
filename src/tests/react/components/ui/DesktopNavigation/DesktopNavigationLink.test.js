/* Packages */
import React from 'react';
import {shallow} from 'enzyme';


/* Component */
import {DesktopNavigationLink} from '../../../../../react/components/ui/DesktopNavigation/DesktopNavigationLink'


/* Fixtures */
import links from '../../../../fixtures/links'


/* Test config */
let component;
beforeEach(() => {
    component = shallow(<DesktopNavigationLink />)
});


/* Tests */
test('Render DesktopNavigationLink component without links', () => {
    expect(component).toMatchSnapshot();
})


test('Render DesktopNavigationLink component with internal link', () => {
    component = shallow(<DesktopNavigationLink {...links[0]}/>)
    expect(component).toMatchSnapshot();
})


test('Render DesktopNavigationLink component with external link', () => {
    component = shallow(<DesktopNavigationLink {...links[1]}/>)
    expect(component).toMatchSnapshot();
})


test('Render DesktopNavigationLink component with internal link and sublink', () => {
    component = shallow(<DesktopNavigationLink {...links[2]}/>)
    expect(component).toMatchSnapshot();
})


test('Render DesktopNavigationLink component with external link with sublink', () => {
    component = shallow(<DesktopNavigationLink {...links[3]}/>)
    expect(component).toMatchSnapshot();
})