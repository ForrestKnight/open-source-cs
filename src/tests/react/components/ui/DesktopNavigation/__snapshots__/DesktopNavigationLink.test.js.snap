// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render DesktopNavigationLink component with external link 1`] = `
<li
  className="DesktopNavigationLink"
>
  <a
    className="DesktopNavigationLink__link"
    href="https://www.google.com"
  >
    External link
  </a>
</li>
`;

exports[`Render DesktopNavigationLink component with external link with sublink 1`] = `
<li
  className="DesktopNavigationLink"
>
  <a
    className="DesktopNavigationLink__link"
    href="https://www.google.com"
  >
    External link with sublink
  </a>
  <ul
    className="DesktopNavigationLink__sublink_list"
  >
    <li
      className="DesktopNavigationLink__sublink"
      key="External sublink"
    >
      <a
        className="DesktopNavigationLink__sublink_link"
        href="https://www.google.com"
      >
        External sublink
      </a>
    </li>
  </ul>
</li>
`;

exports[`Render DesktopNavigationLink component with internal link 1`] = `
<li
  className="DesktopNavigationLink"
>
  <NavLink
    activeClassName="active"
    aria-current="page"
    className="DesktopNavigationLink__link"
    to="/"
  >
    Internal link
  </NavLink>
</li>
`;

exports[`Render DesktopNavigationLink component with internal link and sublink 1`] = `
<li
  className="DesktopNavigationLink"
>
  <NavLink
    activeClassName="active"
    aria-current="page"
    className="DesktopNavigationLink__link"
    to="/"
  >
    Internal link with sublink
  </NavLink>
  <ul
    className="DesktopNavigationLink__sublink_list"
  >
    <li
      className="DesktopNavigationLink__sublink"
      key="Internal sublink"
    >
      <NavLink
        activeClassName="active"
        aria-current="page"
        className="DesktopNavigationLink__sublink_link"
        to="/"
      >
        Internal sublink
      </NavLink>
    </li>
  </ul>
</li>
`;

exports[`Render DesktopNavigationLink component without links 1`] = `
<li
  className="DesktopNavigationLink"
>
  <a
    className="DesktopNavigationLink__link"
  />
</li>
`;
